src/main/perf/TaskParser.java

```diff
@@ -104,12 +104,43 @@ public TaskParser(IndexState state,
   // this pattern doesn't handle all variations of floating numbers, such as .9 , but should be good enough for perf test query parsing purpose
   private final static Pattern combinedFieldsPattern = Pattern.compile(" \\+combinedFields=((\\p{Alnum}+(\\^\\d+.\\d)?,)+\\p{Alnum}+(\\^\\d+.\\d)?)");
 
-  public Task parseOneTask(String line) throws ParseException {
-    return new TaskBuilder(line).build();
+  /**
+   * First pass, parsing from String to some task, may/may not be an UnparsedTask
+   * Called within TaskSource while creating tasks
+   */
+  public Task firstPassParse(String line) throws ParseException {
+    return new UnparsedTask(parseCategory(line));
+  }
+
+  /**
+   * Second pass, parsing from UnparsedTask to some concrete tasks, like SearchTask
+   * May not be called from the same parser that did the first pass
```

Comment:
Hmm what does this mean?

Reply:
Just means that the parser that does the second pass could be different from the parser that do the first pass, so don't try to rely on parser's internal state.